src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/MigrateSubjectNameController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Data.Model.Database;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Unit = GovUk.Education.ExploreEducationStatistics.Common.Model.Unit;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api
+{
+    /**
+     * EES-1816 Temporary controller for copying Subject.Name to ReleaseFiles.Name
+     */
+    [ApiController]
+    [Authorize]
+    public class MigrateSubjectNameController : ControllerBase
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly StatisticsDbContext _statisticsDbContext;
+        private readonly IUserService _userService;
+
+        public MigrateSubjectNameController(
+            ContentDbContext contentDbContext,
+            StatisticsDbContext statisticsDbContext,
+            IUserService userService)
+        {
+            _contentDbContext = contentDbContext;
+            _statisticsDbContext = statisticsDbContext;
+            _userService = userService;
+        }
+
+        [HttpPatch("api/subjects/migrate-subject-names")]
+        public async Task<ActionResult<Unit>> MigrateSubjectNames()
+        {
+            return await _userService.CheckCanRunMigrations()
+                .OnSuccessVoid(async () =>
+                {
+                    await _contentDbContext.ReleaseFiles
+                        .Include(rf => rf.File)
+                        .Where(rf => rf.File.Type == FileType.Data)
```

Comment:
You could add a check here for ReleaseFiles with no Name to make sure you don't overwrite any that are created after the deploy but before this manual endpoint is run. It seems like it might be possible at the moment that after the deploy a new file upload begins and the subject gets created but now without a name, and then the manual migration is run where `rf.Name = subject.Name` will wipe out the name that's been set by the user.